internal/rbd/rbd_util.go

```diff
@@ -556,9 +556,14 @@ func deleteImage(ctx context.Context, pOpts *rbdVolume, cr *util.Credentials) er
 	}
 
 	// attempt to use Ceph manager based deletion support if available
+	// construct the path with pool and rados namespace if present
+	poolPath := pOpts.Pool
+	if pOpts.RadosNamespace != "" {
+		poolPath = poolPath + "/" + pOpts.RadosNamespace
+	}
 	args := []string{
 		"trash", "remove",
-		pOpts.Pool + "/" + pOpts.ImageID,
+		poolPath + "/" + pOpts.ImageID,
```

Comment:
I think it makes sense to put this in a small helper. If `rbdVolume.String()` can not detect it automatically, introduce `GetTrashPath()` or something like it.

Reply:
My main concern is that we're building the image-spec formatted strings in different location, and we have regularly missed updating some. If there is are helper functions that format all image-spec strings, it will be easier to re-use and less likely to make mistakes.

Reply:
added helper function.